client/js/power-zone.js

```diff
@@ -130,43 +130,47 @@ function numZoneHandler(zoneValue) {
   renderZoneCards(zoneValue);
 }
 
+function isNumericString(str) {
+  return !Number.isNaN(str);
+}
+
 // eslint-disable-next-line no-unused-vars
 function formSubmitHandler(event) {
   event.preventDefault();
   const form = document.getElementById('powerZoneForm');
   const outputDict = {};
+  const inputPrefix = 'input';
+  const zonePrefix = 'zone';
   for (let index = 1; index < form.elements.length - 1; index += 1) {
     // Group up 'inputs' information
-    const inputMatch = form.elements[index].id.match(
-      /(?<input>input)(?<value>.*)/,
-    );
-    if (inputMatch) {
+    const { id } = form.elements[index];
+    if (id.startsWith(inputPrefix)) {
       let inputDict = {};
       if (outputDict.inputs) {
         inputDict = outputDict.inputs;
       }
-      inputDict[inputMatch.groups.value] = form.elements[index].value;
+      inputDict[id.substring(inputPrefix.length)] = form.elements[index].value;
```

Comment:
Maybe bring out `id.substring(inputPrefix.length)` into it's own variable to make it more clearer.